tools/autograd/templates/VariableType.h

```diff
@@ -53,6 +53,7 @@ struct VariableType : public at::Type {
   std::tuple<Variable, Variable> as_variable(std::tuple<Tensor, Tensor> tensor) const;
   std::tuple<Variable, Variable, Variable> as_variable(std::tuple<Tensor, Tensor, Tensor> tensor) const;
   std::vector<Variable> as_variable(TensorList tensor) const;
+  Variable maybe_wrap(Tensor data, const Variable & self, bool inplace) const;
```

Comment:
I can't see a reason why would this need to be a `VariableType` method (it doesn't even use `this`) and not a static helper in the file. I think it would be clearer that way

Reply:
It calls `as_variable` which is on `VariableType`. `as_variable` has to be on VariableType because the overload for `Scalar`uses `this->baseType`. C++ doesn't allow a mix of overloads (`as_variable`) that are methods and functions.

Reply:
This can be simplified once ATen only returns "Tensor" instead of "Scalar"